### ISSUE TITLE
v0.3 breaks ShapContributionsGraphComponent for purely categorical data

### ISSUE THREAD

**Issue by hkoppen** [oegedijk/explainerdashboard#78]
After updating to v0.3, my ShapContributionsGraphComponent containing three categorical variables broke (i.e. it shows an empty plot) with the following error:

```
Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "C:\Users\...\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\...\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\...\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\...\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "C:\Users\...\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\...\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\Users\...\site-packages\dash\dash.py", line 1078, in dispatch
    response.set_data(func(*args, outputs_list=outputs_list))
  File "C:\Users\...\site-packages\dash\dash.py", line 1009, in add_context
    output_value = func(*args, **kwargs)  # %% callback invoked %%
  File "C:\Users\...\site-packages\explainerdashboard\dashboard_components\shap_components.py", line 1090, in update_output_div
    pos_label=pos_label, higher_is_better=self.higher_is_better)
  File "C:\Users\...\site-packages\explainerdashboard\explainers.py", line 42, in inner
    return func(self, *args, **kwargs)
  File "C:\Users\...\site-packages\explainerdashboard\explainers.py", line 1332, in plot_contributions
    contrib_df = self.get_contrib_df(index, X_row, topx, cutoff, sort, pos_label)
  File "C:\Users\...\site-packages\explainerdashboard\explainers.py", line 42, in inner
    return func(self, *args, **kwargs)
  File "C:\Users\...\site-packages\explainerdashboard\explainers.py", line 1075, in get_contrib_df
    remove_cat_names(X_row_merged, self.onehot_dict),
  File "C:\Users\...\site-packages\explainerdashboard\explainer_methods.py", line 273, in remove_cat_names
    X_cats.loc[:, cat] = X_cats.loc[:, cat].map(mapping, na_action='ignore').values
  File "C:\Users\...\site-packages\pandas\core\indexing.py", line 670, in __setitem__
    iloc._setitem_with_indexer(indexer, value)
  File "C:\Users\...\site-packages\pandas\core\indexing.py", line 1745, in _setitem_with_indexer
    isetter(ilocs[0], value)
  File "C:\Users\...\site-packages\pandas\core\indexing.py", line 1691, in isetter
    ser._mgr = ser._mgr.setitem(indexer=pi, value=v)
  File "C:\Users\...\site-packages\pandas\core\internals\managers.py", line 524, in setitem
    return self.apply("setitem", indexer=indexer, value=value)
  File "C:\Users\...\site-packages\pandas\core\internals\managers.py", line 396, in apply
    applied = getattr(b, f)(**kwargs)
  File "C:\Users\...\site-packages\pandas\core\internals\blocks.py", line 1684, in setitem
    self.values[indexer] = value
  File "C:\Users\...\site-packages\pandas\core\arrays\categorical.py", line 1992, in __setitem__
    "Cannot set a Categorical with another, "
ValueError: Cannot set a Categorical with another, without identical categories
127.0.0.1 - - [28/Jan/2021 17:52:04] "POST /_dash-update-component HTTP/1.1" 500 -

```

**Comment by oegedijk** [comment]
Ah, you need to update pandas to version `1.2`: `pip install -U pandas` 

(which unfortunately only works with python 3.7+, so if you're using 3.6 you'll also need to upgrade this as well) 

(this is now a dependency for `explainerdashboard 0.3.0.1`

**Comment by oegedijk** [comment]
Think I found a way to make it work with pandas `1.1`, so that would mean that python 3.6 becomes compatible again as well...

**Comment by hkoppen** [comment]
Nice - I'm using python 3.7.4, will test explainerdashboard 0.3.0.1 on monday.

**Comment by oegedijk** [comment]
Just released v0.3.1, think this should be fixes now!

**Comment by hkoppen** [comment]
Yes, thanks!
